### ISSUE TITLE
just_audio_background cannot use in android targetSdkVersion 31

### ISSUE THREAD

**Issue by WingCH** [ryanheise/just_audio#615]
**Which API doesn't behave as documented, and how does it misbehave?**
1. just_audio_background cannot use in android targetSdkVersion 31

**Minimal reproduction project**
Provide a link here using one of two options:
1. just_audio_background example

**To Reproduce (i.e. user steps, not code)**
1. Change `compileSdkVersion` and `targetSdkVersion` to 31 in `just_audio_background/example/android/app/build.gradle`
2. add `android:exported="true"` to fit 31 version requirement
![image](https://user-images.githubusercontent.com/19588517/146865089-4f3701c3-8062-4355-b40b-1ad95e0d94d8.png)

3. Run in android 12 device
4. See error,  the example cannot be open

**Error messages**

```
Launching lib/main.dart on Pixel 4a in debug mode...
Running Gradle task 'assembleDebug'...
✓  Built build/app/outputs/flutter-apk/app-debug.apk.
Installing build/app/outputs/flutter-apk/app.apk...
W/FlutterActivityAndFragmentDelegate(28007): A splash screen was provided to Flutter, but this is deprecated. See flutter.dev/go/android-splash-migration for migration steps.
E/AndroidRuntime(28007): FATAL EXCEPTION: main
E/AndroidRuntime(28007): Process: com.nngnstudio.all, PID: 28007
E/AndroidRuntime(28007): java.lang.RuntimeException: Unable to create service com.ryanheise.audioservice.AudioService: java.lang.IllegalArgumentException: com.nngnstudio.all: Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent.
E/AndroidRuntime(28007): Strongly consider using FLAG_IMMUTABLE, only use FLAG_MUTABLE if some functionality depends on the PendingIntent being mutable, e.g. if it needs to be used with inline replies or bubbles.
E/AndroidRuntime(28007): 	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4500)
E/AndroidRuntime(28007): 	at android.app.ActivityThread.access$1700(ActivityThread.java:247)
E/AndroidRuntime(28007): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2072)
E/AndroidRuntime(28007): 	at android.os.Handler.dispatchMessage(Handler.java:106)
E/AndroidRuntime(28007): 	at android.os.Looper.loopOnce(Looper.java:201)
E/AndroidRuntime(28007): 	at android.os.Looper.loop(Looper.java:288)
E/AndroidRuntime(28007): 	at android.app.ActivityThread.main(ActivityThread.java:7838)
E/AndroidRuntime(28007): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime(28007): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
E/AndroidRuntime(28007): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
E/AndroidRuntime(28007): Caused by: java.lang.IllegalArgumentException: com.nngnstudio.all: Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent.
E/AndroidRuntime(28007): Strongly consider using FLAG_IMMUTABLE, only use FLAG_MUTABLE if some functionality depends on the PendingIntent being mutable, e.g. if it needs to be used with inline replies or bubbles.
E/AndroidRuntime(28007): 	at android.app.PendingIntent.checkFlags(PendingIntent.java:375)
E/AndroidRuntime(28007): 	at android.app.PendingIntent.getActivityAsUser(PendingIntent.java:458)
E/AndroidRuntime(28007): 	at android.app.PendingIntent.getActivity(PendingIntent.java:444)
E/AndroidRuntime(28007): 	at android.app.PendingIntent.getActivity(PendingIntent.java:408)
E/AndroidRuntime(28007): 	at com.ryanheise.audioservice.AudioService.configure(AudioService.java:339)
E/AndroidRuntime(28007): 	at com.ryanheise.audioservice.AudioService.onCreate(AudioService.java:252)
E/AndroidRuntime(28007): 	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4487)
E/AndroidRuntime(28007): 	... 9 more
```

**Expected behavior**
Can open apps

**Screenshots**
--

**Desktop (please complete the following information):**
 - OS: --
 - Browser --

**Smartphone (please complete the following information):**
 - Device: Pixel 4a
 - OS: Android 12

**Flutter SDK version**
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.8.1, on macOS 12.1 21C52 darwin-arm, locale zh-Hant-HK)
```

**Additional context**
--

**Comment by ryanheise** [comment]
> **To Reproduce (i.e. user steps, not code)**
> 
>  1. Change `compileSdkVersion` and `targetSdkVersion` to 31 in `just_audio_background/example/android/app/build.gradle`

Isn't that code, not user steps?

>  2. add `android:exported="true"` to fit 31 version requirement
>        ![image](https://user-images.githubusercontent.com/19588517/146865089-4f3701c3-8062-4355-b40b-1ad95e0d94d8.png)

Ditto.

I wonder how I can make the instruction "**user steps, not code**" clearer so that it won't be misunderstood. Do you have any suggestion on that?

As for the issue, there is already a fix, but until it is released, you'll need to use a dependency override to use the `fix/android_intent_flags` branch of audio_service in git.

**Comment by WingCH** [comment]
Sorry for my misunderstood, maybe you can add some examples of right and wrong in the template.

Anyway thanks for your help!, After I use `fix/android_intent_flags`, android 12 works normally,
```
dependency_overrides:
  audio_service:
    git:
      url: https://github.com/ryanheise/audio_service.git
      path: audio_service/
      ref: 6c29b4cbce37e96977ffaa69a9e0731e63157128
```

**Comment by ryanheise** [comment]
> Sorry for my misunderstood, maybe you can add some examples of right and wrong in the template.

Thanks for that suggestion, I will try it.

**Comment by ryanheise** [comment]
Reopening until the fix is published.

**Comment by ryanheise** [comment]
I have published the fix in `just_audio_background-v0.0.1-beta.3`. Please let me know if that release does not work and I can reopen this issue.
